src/pytest_mock_resources/fixture/database/relational/postgresql.py

```diff
@@ -29,31 +44,26 @@ def create_postgres_fixture(*ordered_actions, **kwargs):
 
     session = kwargs.pop("session", None)
 
+    async_ = kwargs.pop("async_", False)
+
     if len(kwargs):
         raise KeyError("Unsupported Arguments: {}".format(kwargs))
 
     @pytest.fixture(scope=scope)
-    def _(_postgres_container, pmr_postgres_config):
-        database_name = _create_clean_database(pmr_postgres_config)
-        engine = get_sqlalchemy_engine(pmr_postgres_config, database_name)
-
-        assign_fixture_credentials(
-            engine,
-            drivername="postgresql+psycopg2",
-            host=pmr_postgres_config.host,
-            port=pmr_postgres_config.port,
-            database=database_name,
-            username=pmr_postgres_config.username,
-            password=pmr_postgres_config.password,
-        )
+    def _sync(_postgres_container, pmr_postgres_config):
+        engine_manager = create_engine_manager(pmr_postgres_config, ordered_actions, tables)
+        yield from engine_manager.manage_sync(session=session)
 
-        engine_manager = EngineManager(
-            engine, ordered_actions, tables=tables, default_schema="public"
-        )
-        for engine in engine_manager.manage(session=session):
+    @pytest.fixture(scope=scope)
+    async def _async(_postgres_container, pmr_postgres_config):
+        engine_manager = create_engine_manager(pmr_postgres_config, ordered_actions, tables)
+        async for engine in engine_manager.manage_async(session=session):
```

Comment:
could the problem have been that you weren't awaiting or yielding from it? would at the least expect you to need to `yield from engine_manager.manage_async(session=session)` which i would imagine would solve your for-loop needs. but maybe `yield from await engine_manager.manage_async(session=session)` given that it's an async method

Reply:
That wouldn't work. Python does not support `yield from` syntax inside async functions.

Reply:
wat. TIL